File: app/src/main/java/com/hyapp/achat/model/DetailsMessage.java
package com.hyapp.achat.model;

import android.text.format.DateUtils;

public class DetailsMessage extends Message {

    protected String details;

    public DetailsMessage(long timeMillis, String details) {
        super(TYPE_DETAILS, TRANSFER_TYPE_RECEIVE, timeMillis);
        this.details = details;
    }

    public DetailsMessage(long timeMillis) {
        super(TYPE_DETAILS, TRANSFER_TYPE_RECEIVE, timeMillis);
        setDetails(timeMillis);
    }

    public String getDetails() {
        return details;
    }

    public void setDetails(String details) {
        this.details = details;
    }

    public void setDetails(long timeMillis) {
        details = DateUtils.getRelativeTimeSpanString(
                timeMillis
                , System.currentTimeMillis()
                , DateUtils.DAY_IN_MILLIS
                , DateUtils.FORMAT_ABBREV_RELATIVE
        ).toString();
    }
}

File: app/src/main/java/com/hyapp/achat/model/entity/DetailsMessage.kt
package com.hyapp.achat.model.entity

import android.text.format.DateUtils

class DetailsMessage : Message {

    var details: String = ""

    constructor(time: Long, details: String) : super(type = TYPE_DETAILS, time = time) {
        this.details = details
    }

    constructor(time: Long) : super(type = TYPE_DETAILS, time = time) {
        setDetails(time)
    }

    fun setDetails(timeMillis: Long) {
        details = DateUtils.getRelativeTimeSpanString(
                timeMillis, System.currentTimeMillis(), DateUtils.DAY_IN_MILLIS, DateUtils.FORMAT_ABBREV_RELATIVE
        ).toString()
    }
}